### PR TITLE
feat(template): update sdk version and remove loadConfiguration()

### PR DESCRIPTION
**Pull request by tecton**
Update the SDK version and API.
Related PR: #3757 

E2E TEST: Manual test.

### CODE REVIEW

File: templates/function-base/ts/default/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "dependencies": {
         "@azure/functions": "^1.2.2",
         "@microsoft/microsoft-graph-client": "^3.0.0",
-        "@microsoft/teamsfx": "^0.5.0",
+        "@microsoft/teamsfx": "^0.5.2-alpha.594b33666.0",
```
Comment (SLdragon) [inline templates/function-base/ts/default/package.json:16]: Can we use 0.6.0-alpha.xxx?

Comment (tecton) [reply templates/function-base/ts/default/package.json:16]: Yes, I can update the SDK version next week. There's no 0.6.0-alpha version now.

File: templates/tab/ts/default/src/components/sample/Graph.tsx [hunk 2]
```diff
@@ -2,19 +2,20 @@ import "./Graph.css";
...
 export function Graph() {
   const { loading, error, data, reload } = useGraph(
-    async (graph, credential, scope) => {
+    async (graph, teamsfx, scope) => {
       // Call graph api directly to get user profile information
       const profile = await graph.api("/me").get();
 
       // Initialize Graph Toolkit TeamsFx provider
-      const provider = new TeamsFxProvider(credential, scope);
+      const provider = new TeamsFxProvider(teamsfx.getCredential() as TeamsUserCredential, scope);
```
Comment (SLdragon) [inline templates/tab/ts/default/src/components/sample/Graph.tsx:18]: TODO: update this when auth provider is ready

Comment (SLdragon) [reply templates/tab/ts/default/src/components/sample/Graph.tsx]: Updated

**Review by hund030 — APPROVED** [review]

**Pull request merged**